test/caesium/crypto/pwhash_test.clj

```diff
@@ -0,0 +1,149 @@
+(ns caesium.crypto.pwhash-test
+  (:require [caesium.crypto.pwhash :as p]
+            [caesium.byte-bufs :as bb]
+            [caesium.test-utils :refer [const-test]]
+            [caesium.vectors :as v]
+            [caesium.util :as u]
+            [caesium.randombytes :as r]
+            [clojure.test :refer [are deftest is]]))
+
+
+(const-test
+ p/alg-argon2i13 1
+ p/alg-default 1
+ p/bytes-min 16
+ p/bytes-max 4294967295
+ p/passwd-min 0
+ p/passwd-max 4294967295
+ p/saltbytes 16
+ p/strbytes 128
+ p/strprefix "$argon2i$"
+ p/opslimit-min 3
+ p/opslimit-max 4294967295
+ p/memlimit-min 8192
+ p/opslimit-interactive 4
+ p/memlimit-interactive  33554432
+ p/opslimit-moderate 6
+ p/memlimit-moderate 134217728
+ p/opslimit-sensitive 8
+ p/memlimit-sensitive 536870912
+
+ p/argon2i-alg-argon2i13 1
+ p/argon2i-bytes-min 16
+ p/argon2i-bytes-max 4294967295
+ p/argon2i-passwd-min 0
+ p/argon2i-passwd-max 4294967295
+ p/argon2i-saltbytes 16
+ p/argon2i-strbytes 128
+ p/argon2i-strprefix "$argon2i$"
+ p/argon2i-opslimit-min 3
+ p/argon2i-opslimit-max 4294967295
+ p/argon2i-memlimit-min 8192
+ p/argon2i-opslimit-interactive 4
+ p/argon2i-memlimit-interactive  33554432
+ p/argon2i-opslimit-moderate 6
+ p/argon2i-memlimit-moderate 134217728
+ p/argon2i-opslimit-sensitive 8
+ p/argon2i-memlimit-sensitive 536870912
+
+ p/argon2id-alg-argon2id13 2
+ p/argon2id-bytes-min 16
+ p/argon2id-bytes-max 4294967295
+ p/argon2id-passwd-min 0
+ p/argon2id-passwd-max 4294967295
+ p/argon2id-saltbytes 16
+ p/argon2id-strbytes 128
+ p/argon2id-strprefix "$argon2id$"
+ p/argon2id-opslimit-min 1
+ p/argon2id-opslimit-max 4294967295
+ p/argon2id-memlimit-min 8192
+ p/argon2id-opslimit-interactive 2
+ p/argon2id-memlimit-interactive  67108864
+ p/argon2id-opslimit-moderate 3
+ p/argon2id-memlimit-moderate 268435456
+ p/argon2id-opslimit-sensitive 4
+ p/argon2id-memlimit-sensitive 1073741824
+
+ p/scryptsalsa208sha256-bytes-min 16
+ p/scryptsalsa208sha256-bytes-max 0x1fffffffe0
```

Comment:
FWIW this test is failing, the actual value of `p/scryptsalsa208sha256-bytes-max` appears to be -1 on CI.

Reply:
yeah. the "max" ones are the issue. I have noticed that some of them return -1 (even though the include file clearly lists a value). for the sake of this PR, I can pull the failing tests out since these have only to do with internal constants defined by sodium.

Reply:
scrypt_bytes_max is MIN(SIZE_MAX, 0x1fffffffe0). Which might be SIZE_MAX on 32-bit platforms, but in any case, this should used as an unsigned number.

Reply:
You probably shouldn't pull the test out, that just means it's going to bite someone else. The issue is that Java is treating it as a signed number, apparently. I'm still kinda confused, because it's a long and `0x1fffffffe0` is only 37 bits long (so wouldn't fit in a Java int, which is an int32, but should fit fine in a Java long, which is an int64...)

Maybe travis runs on 32 bit? But even then, the long should fit SIZE_MAX (which I'm assuming is 2**32-1?) just fine :) Any thoughts, @jedisct1 ?

Reply:
Not much. Travis appears to be 64 bits, and  `crypto_pwhash_scryptsalsa208sha256_bytes_max()()` return `137438953440` as expected :/

Reply:
@lvh,
I updated the code for version to sodium 1.0.15. Please take a look and tell me what you think.